kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/web/struts/PaymentRequestAction.java

```diff
@@ -0,0 +1,96 @@
+import edu.arizona.kfs.module.purap.PurapConstants;
+import edu.arizona.kfs.module.purap.PurapKeyConstants;
+import edu.arizona.kfs.sys.KFSConstants;
+import edu.arizona.kfs.sys.KFSKeyConstants;
+
+public class PaymentRequestAction extends org.kuali.kfs.module.purap.document.web.struts.PaymentRequestAction {
+    @SuppressWarnings("unused")
+    private static Logger LOG = Logger.getLogger(PaymentRequestAction.class);
+
+    private static transient volatile DisbursementVoucherInvoiceService disbursementVoucherInvoiceService;
+    private static transient volatile ConfigurationService configurationService;
+
+    private static DisbursementVoucherInvoiceService getDisbursementVoucherInvoiceService() {
+        if (disbursementVoucherInvoiceService == null) {
+            disbursementVoucherInvoiceService = SpringContext.getBean(DisbursementVoucherInvoiceService.class);
+        }
+        return disbursementVoucherInvoiceService;
+    }
+
+    private static ConfigurationService getConfigurationService() {
+        if (configurationService == null) {
+            configurationService = SpringContext.getBean(ConfigurationService.class);
+        }
+        return configurationService;
+    }
+
+    @Override
+    protected ActionForward performDuplicatePaymentRequestAndEncumberFiscalYearCheck(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response, PaymentRequestDocument paymentRequestDocument) throws Exception {
+        ActionForward forwardIfDuplicatePREQ = super.performDuplicatePaymentRequestAndEncumberFiscalYearCheck(mapping, form, request, response, paymentRequestDocument);
+        if (forwardIfDuplicatePREQ != null) {
+            return forwardIfDuplicatePREQ;
+        }
+
+        Object question = request.getParameter(KFSConstants.QUESTION_INST_ATTRIBUTE_NAME);
+        Object buttonClicked = request.getParameter(KFSConstants.QUESTION_CLICKED_BUTTON);
+        if (PurapConstants.PREQ_DUPLICATE_DV_QUESTION.equals(question)) {
+            if (ConfirmationQuestion.NO.equals(buttonClicked)) {
+                paymentRequestDocument.getFinancialSystemDocumentHeader().setWorkflowDocumentStatusCode(PurapConstants.PaymentRequestStatuses.APPDOC_INITIATE);
+                String errorText = getConfigurationService().getPropertyValueAsString(PurapKeyConstants.MESSAGE_PREQ_DUPLICATE_DV_INVOICE);
+                // GlobalVariables.getMessageMap().putError(PurapPropertyConstants.INVOICE_NUMBER, PurapKeyConstants.MESSAGE_PREQ_DUPLICATE_DV_INVOICE);
+                return mapping.findForward(KFSConstants.MAPPING_BASIC);
+            } else {
+                return null;
+            }
+        }
+
+        PaymentRequestDocument preq = ((PaymentRequestForm) form).getPaymentRequestDocument();
+        ArrayList<String> matchingDVs = findMatchingDisbursementVouchersWithInvoiceNumber(preq);
+        if (matchingDVs != null && !matchingDVs.isEmpty()) {

```

Comment:
If you have decided to use the DV message constant, please get rid of the PURAP message and associated constant. I would actually prefer that you keep the PURAP version and use that.